### PR TITLE
Feature/homepage-assets

### PR DESCRIPTION
**Pull request by programbo**

### CODE REVIEW

File: apps/frontend/components/BackgroundVideo.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  }
+}
+
+export function BackgroundVideo({ className, enabled, ...options }: BackgroundVideoProps) {
+  const containerElementId = useId()
+  const playerElementId = useId()
+  const active = useSelector(hoverWatcherService, (state) => state.context.activeItem === containerElementId)
+  const ready = useSelector(hoverWatcherService, (state) => state.matches('DOM available'))
+  const youTubeRef = useInterpret(youtubePlayerMachine, {
+    // context: {
```
Comment (jedimonkey) [inline apps/frontend/components/BackgroundVideo.tsx:27]: keep?

File: apps/frontend/components/BackgroundVideo.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
```
Comment (jedimonkey) [inline apps/frontend/components/BackgroundVideo.tsx:27]: need?

File: apps/frontend/components/BackgroundVideos.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    const element = parentElementRef.current
+    if (domAvailable && element && !hasParentElement) {
+      hoverWatcherService.send({ type: 'Set parent element', element })
+    }
+  }, [domAvailable, hasParentElement])
+
+  return (
+    <div ref={parentElementRef} className="absolute top-0 left-0 grid w-full h-screen grid-cols-2">
+      {ids.map((id, index) => (
+        <BackgroundVideo key={`${id}-${index}`} videoId={id} start={15 + index * 15} enabled={enabled} />
```
Comment (jedimonkey) [inline apps/frontend/components/BackgroundVideos.tsx:34]: is id unique enough? maybe that should be used for the key alone?

File: apps/frontend/components/HeroWithNavigation/MainNavigation.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    <motion.ul
+      className="flex items-center h-10 px-10"
+      variants={navVariants}
+      initial="out"
+      animate={mode}
+      onMouseLeave={handleMouseLeave}>
+      {navigationItems.map(({ id, title, uri }) => (
+        <motion.li className="relative flex flex-col justify-center px-3" key={id} variants={itemVariants}>
+          <Link href={uri}>
+            <a
```
Comment (jedimonkey) [inline apps/frontend/components/HeroWithNavigation/MainNavigation.tsx:72]: ... didn't updated link remove the need for <a> ?

File: apps/frontend/components/HeroWithNavigation/SecondaryNavigation.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  return (
+    <div className={clsx('', final ? 'ml-0' : '-ml-12')}>
+      <motion.div className={clsx('grid grid-cols-2 uppercase', final ? 'gap-x-64' : 'gap-x-11')}>
+        <SecondaryNavigationList
+          variants={listVariants}
+          title="Buying"
+          className={clsx('font-display', final ? 'items-start' : 'items-end')}>
+          <SecondaryNavigationItem
+            index={0}
+            href={'/'}
```
Comment (jedimonkey) [inline apps/frontend/components/HeroWithNavigation/SecondaryNavigation.tsx:36]: stubbed I presume for now?

File: apps/frontend/components/HeroWithNavigation/SecondaryNavigationItem.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  children: string
+  href: string
+  alignment: 'left' | 'right'
+}
+
+export function SecondaryNavigationItem({
+  index,
+  width,
+  height,
+  d,
```
Comment (jedimonkey) [inline apps/frontend/components/HeroWithNavigation/SecondaryNavigationItem.tsx:24]: oh, you can pass in the data for the svg. I was wondering wth 'd' was....

File: apps/frontend/components/HeroWithNavigation/index.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+export type HeroPhase = 'Start' | 'Middle' | 'End'
+export type HeroPhases = Record<HeroPhase, Variant>
+
+interface HeroWithNavigationProps extends MainNavigationProps {
+  phoneNumber: string
+}
+
+const homeHero = interpret(homeHeroMachine).start()
+export const HomeHeroContext = createContext<ActorRefFrom<typeof homeHeroMachine>>(homeHero)
```
Comment (jedimonkey) [inline apps/frontend/components/HeroWithNavigation/index.tsx:25]: maybe we should move the context out of here?

File: apps/frontend/components/HeroWithNavigation/index.tsx [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+          {!completed && (
+            <motion.div initial={{ y: 0 }} animate={{ y: creeping ? -50 : 0 }} transition={{ duration: 4.5 }}>
+              <Logo />
+            </motion.div>
+          )}
+          {showSecondaryNavigation && <SecondaryNavigation />}
+        </motion.div>
+
+        {!completed && (
+          <style jsx global>{`
```
Comment (jedimonkey) [inline apps/frontend/components/HeroWithNavigation/index.tsx:84]: heh, kinda cute way to switch between overflow

File: apps/frontend/machines/homeHero.machine.ts [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+  showDuality: boolean
+  activePanel: VideoPanels | null
+  videoPanels: Record<VideoPanels, HTMLDivElement | null>
+  videoPlayers: Record<VideoPanels, YT.Player | null>
+  videoRefs: Record<VideoPanels, ActorRefFrom<typeof youtubeApiMachine> | null>
+}
+
+type Events = { type: 'Reset' }
+
+// const isHovering = (panel: HTMLDivElement | null) => !!panel && panel.parentNode?.querySelector(':hover') === panel
```
Comment (jedimonkey) [inline apps/frontend/machines/homeHero.machine.ts:37]: remove?

File: apps/frontend/pages/index.tsx [hunk 10]
```diff
@@ -1,21 +1,29 @@
+// import { inspect } from '@xstate/inspect'
```
Comment (jedimonkey) [inline apps/frontend/pages/index.tsx:1]: ?

File: apps/frontend/pages/index.tsx [hunk 11]
```diff
@@ -1,21 +1,29 @@
...
+  const navigationItems = [...navigationNodes.map(({ element: { id, title, uri } }) => ({ id, title, uri }))]
   return (
     <div className="w-full min-h-screen">
       <Head>
         <title>DUET Property</title>
         <meta name="description" content="Generated by create next app" />
         <link rel="icon" href="/favicon.ico" />
       </Head>
-      <header className="flex justify-center h-24 bg-black">
+      {/* <header className="flex justify-center h-24 bg-zinc-150">
```
Comment (jedimonkey) [inline apps/frontend/pages/index.tsx:26]: remove or keep?

File: apps/frontend/components/HeroWithNavigation/index.tsx (hunk 7 above)
Comment (programbo) [reply apps/frontend/components/HeroWithNavigation/index.tsx:25]: It is but you can also interpret and start them in their own module and import them anywhere. That's what I was doing previously but put it into a context whilst problem solving some issues that turned out to be unrelated.

File: apps/frontend/components/BackgroundVideos.tsx (hunk 3 above)
Comment (programbo) [reply apps/frontend/components/BackgroundVideos.tsx:34]: I initially only had one video :D

File: apps/frontend/machines/homeHero.machine.ts [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+        //   }
+        // }),
+      },
+      guards: {
+        // 'Hovering over Selling': ({ videoPanels }, _evt) => isHovering(videoPanels.selling),
+        // 'Hovering over Buying': ({ videoPanels }, _evt) => isHovering(videoPanels.buying),
+        // 'Missing video information': (_ctx, { name, videoId, elementId }) => !name || !videoId || !elementId,
+      },
+      services: {
+        // 'Watch panels':
```
[the remark this answers is not in the document]
Comment (programbo) [reply apps/frontend/machines/homeHero.machine.ts:132]: Machine got too big and was handling two distinct behaviours so I split it up.

**Pull request merged**